docs(file-system): add working bundled asset examples using expo asset

dileepapeiris opened the pull request:
### Summary

Closes #41267.

This PR addresses confusion regarding the `bundleDirectory` property in `expo-file-system`. Users were previously led to believe that `bundleDirectory` automatically provides access to assets located in the JavaScript project structure, whereas it actually points to the **native platform bundle** (e.g., Android assets folder or iOS main bundle).

The documentation has been updated to clarify that using `expo-asset` is the recommended approach for JS assets. It also provides explicit guides, prerequisites (moving files to native directories), and code examples for advanced users who need to access the native bundle directly using both the modern `File` API and the legacy API.

### Changes

**File updated:** `filesystem.mdx`

Added a new `<Collapsible summary="Working with Bundled Assets">` section. This section details:
1.  **Using `expo-asset` (Recommended):** A guide on resolving and reading assets imported via JS.
2.  **Using `Paths.bundle` (Advanced / Native only):** Clear instructions on how to move files to the native bundle (via Config Plugins/Xcode/Android Studio), a warning about iOS read-only restrictions, and a code example demonstrating how to copy files to `Paths.document` on iOS before reading.

<img width="1192" height="518" alt="image" src="https://github.com/user-attachments/assets/e42a629b-c5f1-40c0-9f90-ba5376ff238a" />
<img width="1103" height="682" alt="image" src="https://github.com/user-attachments/assets/111e8c50-7b09-4e18-9e81-fad0308a9a1a" />
<img width="1095" height="757" alt="image" src="https://github.com/user-attachments/assets/5b9291f1-d5b7-4d26-9c4d-281016cbd8a1" />
<img width="1119" height="919" alt="image" src="https://github.com/user-attachments/assets/954749be-7777-4b2a-a312-d7dc032eeebb" />


**File updated:** `filesystemlegacy.mdx`

Added a new section `### Reading files from the native bundle`. This section details:
1.  Clarification that `FileSystem.bundleDirectory` requires files to exist in the native platform project.
2.  Prerequisites for Android (`android/app/src/main/assets/`) and iOS (Copy Bundle Resources).
3.  A complete code example demonstrating how to check for file existence and handle the necessary copy operation on iOS (due to sandboxing) using the legacy `FileSystem` API.

<img width="1133" height="698" alt="image" src="https://github.com/user-attachments/assets/10bc3a6c-4eef-4672-9434-fc01e89c628a" />
<img width="1130" height="933" alt="image" src="https://github.com/user-attachments/assets/adbdeb29-1435-413d-92bd-cac6c8a16f6b" />


#### Test Plan

- [x] Reviewed the changes locally to ensure the Markdown renders correctly.
- [x] Checked that code blocks are syntax-highlighted and formatted correctly.
- [x] Verified that the `<Collapsible>` component functions as expected.
- [x] Verified that the explanation clearly distinguishes between `expo-asset` usage and raw native bundle access.
- [x] **Verified Code Examples:** I created two dedicated reproduction repositories to validate the code snippets for both the Modern and Legacy APIs.

    **1. Modern API (`expo-file-system`)**
    - **Repo:** [expo-file-system-bundled-pr-verification](https://github.com/dileepapeiris/expo-file-system-bundled-pr-verification)
    - Validated that `expo-asset` works out-of-the-box.
    - Validated that `Paths.bundle` works on Android (direct access) and iOS (copy-then-read) after manual native setup.

    **2. Legacy API (`expo-file-system/legacy`)**
    - **Repo:** [expo-file-system-legacy-bundled-pr-verification](https://github.com/dileepapeiris/expo-file-system-legacy-bundled-pr-verification)
    - Validated `FileSystem.bundleDirectory` usage.
    - Confirmed the necessity of `copyAsync` for iOS bundle assets versus direct reading on Android.

dileepapeiris committed:
Document FileSystem.bundleDirectory and asset access

Adds documentation for FileSystem.bundleDirectory, explaining its read-only nature and platform-specific details. Provides usage examples for accessing bundled assets using expo-asset and clarifies the need to register assets for proper URI resolution. Notes on Metro's JSON handling and the requirement to copy bundled files before modification are also included.

dileepapeiris committed:
Add documentation for reading bundled assets

Added a new section explaining how to access read-only bundled assets using `Paths.bundle` and the `expo-asset` and `expo-file-system` APIs. The section includes platform-specific notes, usage examples, and instructions for bundling assets in the app build process.

dileepapeiris committed:
Document read-only access to bundled assets in FileSystem

Adds documentation for FileSystem.bundleDirectory, explaining how to access bundled assets, platform-specific details, and usage with expo-asset. Includes code examples and notes on read-only restrictions and proper asset registration.

dileepapeiris committed:
Add documentation for reading bundled assets

Added a new collapsible section explaining how to access read-only bundled assets using `Paths.bundle` and the `expo-asset` and `expo-file-system` APIs. The section includes platform-specific notes, usage examples, and instructions for bundling assets with the app.

chrfalch commented:
Thanks for submitting this PR, @dileepapeiris! Really appreciate it. Could you take a look and see if you can clean up the warnings and get it to pass the docs-pr CI test?

dileepapeiris committed:
Update filesystem docs for bundled asset access

Reorganizes and expands documentation on reading bundled assets in expo-file-system. Adds clearer guidance on using expo-asset (recommended) and direct native bundle access, with updated code examples and platform-specific notes. Also moves the moving/copying files example to its own section for clarity.

dileepapeiris committed:
Revise filesystem docs: clarify bundled asset access

Reorganizes and expands documentation on reading bundled assets in expo-file-system. Adds a new section on moving and copying files, clarifies the recommended use of expo-asset for accessing bundled files, and provides detailed instructions and code samples for both expo-asset and direct native bundle access. Improves platform-specific guidance and updates examples for better clarity.

dileepapeiris committed:
Remove bundled asset access documentation

Removed documentation and code examples related to accessing bundled assets via FileSystem.bundleDirectory and expo-asset. This streamlines the page and omits details about read-only bundled asset access.

dileepapeiris committed:
Remove bundled asset access documentation

Removed documentation and examples related to accessing bundled assets via FileSystem.bundleDirectory and expo-asset. This streamlines the page and removes details about read-only bundled asset access.

dileepapeiris committed:
Add guide for reading files from native bundle

Documents how to access files bundled with the app using FileSystem.bundleDirectory, including platform-specific instructions and example code for both Android and iOS. Clarifies prerequisites for making files available in the native bundle and explains the need to copy files to a writable location on iOS.

dileepapeiris committed:
Add guide for reading files from native bundle

Added documentation explaining how to access files bundled with the app using FileSystem.bundleDirectory. Includes platform-specific instructions for Android and iOS, prerequisites for copying files into the native bundle, and a usage example demonstrating how to read bundled files.

dileepapeiris committed:
Update iOS file copy example in filesystem docs

Corrects the recommended writable location from `Paths.cache` to `Paths.document` in the usage example and updates the code to copy files to the document directory instead of the cache directory on iOS.

dileepapeiris committed:
Update iOS file copy example in filesystem docs

Corrects the recommended writable location from `Paths.cache` to `Paths.document` in the iOS usage example and updates the code to use `Paths.document` for copying files. This clarifies best practices for handling files in the iOS sandbox environment.

dileepapeiris committed:
Clarify asset bundling for Expo managed workflow

Added instructions for configuring assetBundlePatterns in app.json for Expo managed workflow to ensure assets are bundled correctly. Clarified platform-specific requirements and updated notes on using Config Plugins.

dileepapeiris committed:
Update asset bundling instructions for Expo workflows

Clarifies asset bundling requirements for Expo managed and bare workflows in the filesystem documentation. Adds explicit instructions for configuring `assetBundlePatterns` in `app.json` for managed workflow and highlights platform-specific behavior.

dileepapeiris committed:
Clarify asset bundling for FileSystem.bundleDirectory

Updated documentation to distinguish asset bundling steps for Expo managed and bare workflows. Added explicit app.json configuration for managed workflow and clarified platform-specific requirements and notes for iOS and Android.

dileepapeiris committed:
Clarify asset bundling for FileSystem.bundleDirectory

Updated documentation to distinguish asset bundling steps for Expo managed and bare workflows. Added explicit app.json configuration for managed workflow and clarified platform-specific requirements and notes for iOS and Android.

amandeepmittal reviewed (changes requested):
Thank you for creating this PR to add examples with appropriate explanation.
Please see my suggestions below and apply them to SDK 54 versioned docs as well.

dileepapeiris committed:
Update heading level for prerequisites section

Changed the 'Prerequisites' section from bold text to a level 4 heading for improved documentation structure and consistency.

dileepapeiris committed:
Update asset bundling instructions for CNG

Revised the documentation to specify asset bundling steps for Expo projects using CNG, replacing the previous reference to the managed workflow.

dileepapeiris committed:
Update filesystem legacy docs for React Native config

Added usage instructions for the ConfigReactNative component and clarified file placement for Android and iOS in existing React Native apps. Improves guidance for asset configuration in bare workflow scenarios.

dileepapeiris committed:
Update filesystem-legacy docs for clarity and usage

Refines the usage section in the filesystem-legacy documentation by removing outdated config plugin instructions, clarifying code comments, and specifying the file name in the code block for better guidance.

dileepapeiris committed:
Clarify note on accessing files in native bundle

Rewords and reorganizes the note about accessing files from the native platform bundle, emphasizing the need to move files during the build process and linking to the relevant config plugin documentation.

dileepapeiris committed:
Update filesystem docs for asset and bundle usage

Refined instructions for reading files using expo-asset and native bundle access. Clarified prerequisites, platform-specific steps, and updated terminology for Expo and React Native workflows.

dileepapeiris committed:
Update collapsible summary capitalization in docs

Changed 'Working with Bundled Assets' to 'Working with bundled assets' for consistency in documentation formatting.

dileepapeiris committed:
Update filesystem.mdx

dileepapeiris committed:
Update filesystem legacy docs for bundle file access

Clarified instructions for accessing files in the native bundle, added details about config requirements for Expo and React Native projects, and improved code and usage explanations. Introduced ConfigReactNative component for platform-specific setup guidance.

dileepapeiris committed:
Update filesystem docs for bundled asset usage

Improves documentation for working with bundled assets in expo-file-system. Clarifies usage of expo-asset and native bundle access, updates examples, and adds configuration details for React Native and Expo projects.

dileepapeiris committed:
Fix import path for ConfigReactNative component

Updated the import path for the ConfigReactNative component to use the correct location in the filesystem-legacy documentation. Also made a minor formatting adjustment to the Android assets path description.

dileepapeiris committed:
Fix import path for ConfigReactNative component

Updated the import path for the ConfigReactNative component to use the correct location in the filesystem-legacy documentation. Also fixed minor spacing in the Android assets path description.

dileepapeiris committed:
Remove unused imports from filesystem SDK docs

Cleaned up the import statements in the filesystem.mdx documentation file by removing unused components from ConfigSection. This helps maintain code clarity and reduces unnecessary imports.

dileepapeiris committed:
Update heading capitalization for expo-asset section

Changed the heading from 'Using expo-asset (Recommended)' to 'Using expo-asset (recommended)' to maintain consistent capitalization style in documentation.

dileepapeiris committed:
Update heading capitalization for expo-asset section

Changed the heading from 'Using expo-asset (Recommended)' to 'Using expo-asset (recommended)' to maintain consistent capitalization style in documentation.

amandeepmittal reviewed (approved):
Thanks for the docs updates! Looks good!

Let's wait for @aleqsio's technical review on this.

dileepapeiris commented:
Hi @amandeepmittal , @aleqsio is there any update on this ? Thanks

dileepapeiris committed:
Merge branch 'main' into docs-search

dileepapeiris commented:
> @copilot resolve the merge conflicts in this pull request

@brentvatne Resolved
